Frameworks/Foundation/_NSUndoGroup.mm

```diff
@@ -0,0 +1,93 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard.h>
+#import <Foundation/NSUndoManager.h>
+#import "_NSUndoManagerInternal.h"
+
+@implementation _NSUndoGroup {
+    StrongId<NSMutableArray*> _undoGrouping;
```

Comment:
We will however have to traverse this "stack" for undo operations on particular targets though with the -removeAllActionsWithTarget:(id)target method. I figured it would be significantly worse to use std::stack in that situation.

---
In reply to: [76124040](https://github.com/Microsoft/WinObjC/pull/900#discussion_r76124040) [](ancestors = 76124040)